volestipy/volestipy/volestipy.pyx

```diff
@@ -5,51 +5,67 @@
+
+
+# with respect to the "compute_volume" def
+volume_methods = ["sequence_of_balls".encode("UTF-8"), "cooling_gaussian".encode("UTF-8"), "cooling_balls".encode("UTF-8")]
+walk_methods = ["uniform_ball".encode("UTF-8"), "CDHR".encode("UTF-8"), "RDHR".encode("UTF-8"), "gaussian_ball".encode("UTF-8"), "gaussian_CDHR".encode("UTF-8"), "gaussian_RDHR".encode("UTF-8"), "uniform_ball".encode("UTF-8"), "billiard".encode("UTF-8")]
+
 
-volume_methods = ["sequence_of_balls".encode("UTF-8"), "gaussian_annealing".encode("UTF-8")]
 cdef class HPolytope:
    cdef HPolytopeCPP polytope_cpp
    cdef double[:,::1] _A
    cdef double[::1] _b
-
+   
    def __cinit__(self, double[:,::1] A, double[::1] b):
       self._A = A
       self._b = b
-      # assert n_hyperplanes==b.shape[0]
-      # Do we also need to pass strdie to make it more error proof?
-      # https://stackoverflow.com/questions/34592961/passing-multidimensional-memoryviews-to-c-function
       n_hyperplanes, n_variables = A.shape[0], A.shape[1]
       self.polytope_cpp = HPolytopeCPP(&A[0,0], &b[0], n_hyperplanes, n_variables)
 
-   def compute_volume(self, int walk_len=2, double epsilon=0.05, method="sequence_of_balls", np.npy_int32 seed=get_time_seed()):
-      method = method.encode("UTF-8")
-      if method  in volume_methods:
-         return self.polytope_cpp.compute_volume(method, walk_len, epsilon, seed)
+   def compute_volume(self, walk_len=2, epsilon=0.05, vol_method="sequence_of_balls", walk_method="uniform_ball", np.npy_int32 seed=get_time_seed()):
+      vol_method = vol_method.encode("UTF-8")
+      walk_method = walk_method.encode("UTF-8")
+      if vol_method  in volume_methods:
+         if walk_method in walk_methods:
+            return self.polytope_cpp.compute_volume(vol_method, walk_method, walk_len, epsilon, seed)
+         else:
+            raise Exception('"{}" is not implemented to walk methods. Available methods are: {}'.format(walk_method, walk_methods))
       else:
-         raise Exception('"{}" is not implemented to compute volume. Possbile methods are: {}'.format(method, volume_methods))
+         raise Exception('"{}" is not implemented to compute volume. Available methods are: {}'.format(vol_method, volume_methods))
 
-   def generate_samples(self, int walk_len=2, int n_samples=1000, np.npy_int32 seed=get_time_seed()):
-         n_variables = self._A.shape[1]
-         cdef double[:,::1] samples = np.zeros((n_samples,  n_variables), dtype=np.float64, order="C")
-         self.polytope_cpp.generate_samples(walk_len, n_samples, &samples[0,0], seed)
-         return np.asarray(samples)
 
+# we need to build a Python function for getting a starting point depending on the polytope 
+   def generate_samples(self, walk_len=1, number_of_points=1000, number_of_points_to_burn=100, boundary=True, cdhr=True, rdhr=False, gaussian=False, set_L=False, billiard=False, ball_walk=False, a=0, L=0):
```

Comment:
set default boundary = false, and default cdhr=true

Reply:
boundary changed 
cdhr was true